### PR TITLE
feat(#174): bastion Dockerfile and compose file

### LINKED ISSUE #174: Access to  containerized instances of postgres
**Is your feature request related to a problem? Please describe.**
CHT Sync deployments that use the [built in postgres container](https://github.com/medic/cht-sync/blob/main/docker-compose.postgres.yml) in either [Docker](https://docs.communityhealthtoolkit.org/hosting/analytics/setup-docker-compose/) or [Kubernetes](https://docs.communityhealthtoolkit.org/hosting/analytics/setup-kubernetes/) (k8s) will need an easy way allow remote access to postgres server for either external dashboards (eg superset) or end users looking to do data analysis.   

In both k8s and docker, postgres service is not remotely accessible, an intentional secure-by-default design. 

**Describe the solution you'd like**
A [sidecar service](https://kubernetes.io/docs/concepts/workloads/pods/sidecar-containers/), which can be used in both Kubernetes and Docker, would be a great way to allow easy access to postgres.  The sidecar would likely expose SSH, or some other tunneling service, for which an initial connection could be made (eg an SSH tunnel) and then postgres connections could be made on top of/through this.

**Describe alternatives you've considered**
Ephemeral access could be granted by admins who can run `kubectl` to create an account knowing it will get deleted when the container gets rebooted.  Another solution might be to set up an `autossh` sidecar that exposes the postgres service on some other, external infrastracture.

**Additional context**
Medic is currently deploying CHT Sync [internally for MoH Togo](https://github.com/medic/config-moh-togo/issues/327).  Instead of using the deprecated,  monolithic install of Postgres which has old couch2pg running directly on it, we'd like to use a more modern architecture where each CHT Core deployment gets it's own postgres and CHT Sync instance.  This new style of deployment makes it very hard to do both persistent and ad hoc  sql queries.  This will be greatly simplified if we provide an out of the box solution which is both easy to configure and easy to deploy.

### PR DESCRIPTION
**Pull request by mrjones-plip**
# Description

This PR: 
* Adds new `Dockerfile` to build a bastion container - based on Alpine `3.20`
* adds compose file `docker-compose.bastion.yml` to define a bastion host to allow SSH tunnels to access postgres
* locks down postgres by removing the `ports:` lines from docker file so it doesn't expose the port externally (unless you run  pgadmin compose file) 
* splits off pgadmin into it's own compose file. most of the time this shouldn't be run alongside postgres because it exposes a publicly accessible HTTP port that with no password set. While you do need to know the `POSTGRES_PASSWORD`, I think this is still to "insecure by default" so we should silo them off by default instead
* adds `ports:` file to pgadmin compose file for postgres server
* defaults all e2e tests to go through bastion host, just as it will in production
 
closes #174

# Code review checklist

- [x] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [x] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [x] Tested: Unit and/or e2e where appropriate
- [x] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by mrjones-plip** [comment]
@witash and @njuguna-n - is it OK to remove these [two lines](https://github.com/medic/cht-sync/blob/bbe9801b678761165262d0f1325007fc9eae31cf/docker-compose.postgres.yml#L5)? for a sane default for production I don't think we want this exposed - I've [removed it](https://github.com/medic/cht-sync/pull/177/commits/d6eebc138836a479d5e744f82d790edced291a45) in this PR.  

Related: For development I think we should use pgadmin.  I also think we should break out pgadmin to it's own docker file so it's not run by default in production.

**Comment by mrjones-plip** [comment]
@njuguna-n or @witash - before I send this PR out for formal review - can you point me toward how I should add tests for the new bastion container?  Maybe an existing pattern we have?  I'm a test n00b and need a little guidance :pray: 

thanks!

**Comment by njuguna-n** [comment]
Hello @mrjones-plip for this one you are looking at adding e2e tests right? You can have a look at existing e2e tests [here](https://github.com/medic/cht-sync/blob/main/tests/e2e-test.spec.js) for some guidance.

**Comment by mrjones-plip** [comment]
While ready for review, I think we should wait until #187 is merged.  Then I can get all a green CI on this branch - otherwise right now green CI is impossible because of #186 

@lorerod and @dianabarsan - if you can't wait, you can have a preliminary look at e2d and new bastion container respectively! Otherwise, feel free to wait until this PR is fully baked.

**Comment by mrjones-plip** [comment]
hmmm - I note [cleanup](https://github.com/medic/cht-sync/pull/177/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L10) never runs, it just hangs here:

```
 Container cht-sync-postgres-1  Started
      ✔ should handle PostgreSQL downtime gracefully (26937ms)
    Incremental sync
      ✔ should process document edits (18288ms)
      1) should process contact deletes
      ✔ should process person deletes (18039ms)
      ✔ should process report deletes (12023ms)
      ✔ should process incremental inserts (18023ms)
    Conflict resolution
Failed to edit document with id 00172650-a2d1-540b-b308-d53ff75f102c: Document update conflict.
      ✔ should handle update conflicts (19058ms)
Failed to delete document with id 00172650-a2d1-540b-b308-d53ff75f102c: Document update conflict.
      ✔ should handle delete conflicts (19039ms)


  15 passing (3m)
  1 failing

  1) Main workflow Test Suite
       Incremental sync
         should process contact deletes:

      AssertionError: expected 1 to equal +0
      + expected - actual

      -1
      +0
      
      at Proxy.assertEqual (node_modules/chai/lib/chai/core/assertions.js:1038:12)
      at Proxy.methodWrapper (node_modules/chai/lib/chai/utils/addMethod.js:57:25)
      at doAsserterAsyncAndAddThen (node_modules/chai-as-promised/lib/chai-as-promised.js:289:22)
      at Proxy.<anonymous> (node_modules/chai-as-promised/lib/chai-as-promised.js:255:20)
      at Proxy.overwritingMethodWrapper (node_modules/chai/lib/chai/utils/overwriteMethod.js:78:33)
      at Proxy.<anonymous> (node_modules/chai-exclude/chai-exclude.js:122:21)
      at Proxy.overwritingMethodWrapper (node_modules/chai/lib/chai/utils/overwriteMethod.js:78:33)
      at Context.<anonymous> (file:///home/mrjones/Documents/MedicMobile/cht-sync/tests/e2e-test.spec.js:221:49)
      at process.processTicksAndRejections (node:internal/process/task_queues:105:5)

```

Maybe I'm not being patient enough? !

File: tests/utils/bastion-ssh-key-private
Comment (lorerod) [inline tests/utils/bastion-ssh-key-private:1]: Could we explore dynamic secret generation or docker secrets instead? We can do it on a separate PR if you think it is a good idea.

File: tests/e2e-test.spec.js
Comment (lorerod) [inline tests/e2e-test.spec.js:70]: 🤩

**Review by lorerod — APPROVED** [review]
Thank you @mrjones-plip! 
And thanks for also working in the documentation of this change.

File: tests/e2e-test.spec.js
Comment (mrjones-plip) [reply tests/e2e-test.spec.js:70]: Thanks for the review!

I'll resolve this compliment (thanks!) and then leave the PR open for another day or so in case @njuguna-n or @witash have any comments.

**Pull request merged**
